File: palindrome.py
def isPalidrome(x):
    reversedNumber = 0
    number = x 
    
    while number > 0:
        digit = number % 10
        reversedNumber = reversedNumber * 10 + digit
        number //= 10
    if reversedNumber == x:
        return True
    return reversedNumber, False 

x = int(input("x="))
rr = isPalidrome(x)
print(rr)
# print(6789//10)

File: palindrome.cs
using System;

bool isPalindrome(int n){
    int reversedNumber = 0;
    int number = n;

    while(number > 0){
        int digit = number %10;
        reversedNumber =reversedNumber *10 + digit;
        number /= 10;
    }
    if (reversedNumber == n){
        return true;
    }
    return false;
} 

Console.WriteLine(isPalindrome(2002));